docs/data/base/components/checkbox/UnstyledCheckboxIndeterminateGroup.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { styled } from '@mui/system';
+import { Checkbox as BaseCheckbox } from '@mui/base/Checkbox';
+import HorizontalRule from '@mui/icons-material/HorizontalRule';
+import Check from '@mui/icons-material/Check';
+
+const colors = ['Red', 'Green', 'Blue'];
+
+export default function UnstyledCheckboxIndeterminateGroup() {
+  const [checkedValues, setCheckedValues] = React.useState([false, true, false]);
+
+  const isChecked = checkedValues.every((value) => value);
+  const isIndeterminate = checkedValues.some((value) => value) && !isChecked;
+
+  const id = React.useId();
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column' }}>
+      <ListRoot>
+        <Checkbox
+          id={id}
+          aria-controls={colors.map((color) => `${id}-${color}`).join(' ')}
+          indeterminate={isIndeterminate}
+          checked={isChecked}
+          onChange={(event) => {
+            const checked = event.target.checked;
+            setCheckedValues([checked, checked, checked]);
+          }}
+        >
+          <Indicator>
+            {isIndeterminate ? <HorizontalRuleIcon /> : <CheckIcon />}
+          </Indicator>
+        </Checkbox>
+        <Label htmlFor={id} onMouseDown={(e) => e.preventDefault()}>
```

Comment:
We avoid using shorthands like `e` in the codebase, especially for the docs as it makes it not very clear to entry-level developers what the variable is about. 

```suggestion
        <Label htmlFor={id} onMouseDown={(event) => event.preventDefault()}>
```

Fixed in 25eb16b271c6e704bb261bb4df71b6951c0a5585 to match with the rest of the codebase.